services/meta/data.go

```diff
@@ -292,8 +292,11 @@ func (data *Data) DropShard(id uint64) {
 					data.Databases[dbidx].RetentionPolicies[rpidx].ShardGroups[sgidx].Shards = append(shards[:found], shards[found+1:]...)
 
 					if len(shards) == 1 {
-						// We just deleted the last shard in the shard group.
-						data.Databases[dbidx].RetentionPolicies[rpidx].ShardGroups[sgidx].DeletedAt = time.Now()
+						// We just deleted the last shard in the shard group, but make sure we don't overwrite the timestamp if it
+						// was already deleted.
+						if !data.Databases[dbidx].RetentionPolicies[rpidx].ShardGroups[sgidx].Deleted() {
```

Comment:
Is this the key change?  Don't overwrite the `DeletedAt` time?

Reply:
This is not the key change, but seems reasonable.

Reply:
This is necessary to prevent a new bug from appearing in enterprise. Without this change, the enterprise code for nodes to drop references to the shard will cause the shard files to last double the RP duration.